tests/conformance/scripts/run-tests/index.ts

```diff
@@ -131,26 +124,45 @@ async function runWriterTest(
       hadError = true;
       continue;
     }
-    const outputHex = bytesToHex(output);
-    const expectedOutputHex = bytesToHex(expectedOutput as Uint8Array);
-    if (outputHex !== expectedOutputHex) {
-      console.error(colors.red("fail       "), filePath);
-      let colorDiff = "";
-      const charDiff = Diff.diffChars(expectedOutputHex, outputHex);
-      charDiff.forEach((part) => {
-        const color =
-          part.added === true ? colors.green : part.removed === true ? colors.red : colors.grey;
-        colorDiff += color(part.value);
+    const expectedParts = splitMcapRecords(expectedOutput).map(spaceHexString).join("\n");
+    const outputParts = splitMcapRecords(output).map(spaceHexString).join("\n");
+    if (expectedParts !== outputParts) {
+      try {
+        const expectedOutputJson = await fs.readFile(filePath, { encoding: "utf-8" });
+        const reader = new Mcap0StreamReader({ validateCrcs: true });
+        reader.append(output);
+        const records: Mcap0Types.TypedMcapRecord[] = [];
+        for (let record; (record = reader.nextRecord()); ) {
+          records.push(record);
+        }
+
+        const actualOutputJson = stringifyRecords(records, variant);
+        const outputNorm = normalizeJson(actualOutputJson);
+        const expectedNorm = normalizeJson(expectedOutputJson);
+        if (outputNorm !== expectedNorm) {
+          console.error(Diff.createPatch(filePath, expectedNorm, outputNorm));
+        }
+      } catch (err) {
+        // Don't show JSON if we can't parse it.
```

Comment:
I think it's useful to output the error, in case it helps determine why the file is not valid

Reply:
Ok I've put it back.